src/users_db.rs

```diff
@@ -111,34 +111,34 @@ impl UserBuilder {
         self
     }
 
-    pub fn set_admin(&mut self, admin: bool) -> &mut UserBuilder {
+    pub fn set_admin(mut self, admin: bool) -> Self {
         self.is_admin = Some(admin);
         self
     }
 
-    pub fn finalize(&mut self) -> Result<User, UserWithError> {
+    pub fn finalize(mut self) -> Result<User, UserWithError> {
         use rand;
         if self.secret.is_empty() {
-            self.secret(&rand::random::<i32>().to_string());
+            self.secret = rand::random::<i32>().to_string();
```

Comment:
As said, moving mutable values instead of mutable references works fine unless you try to call one of the methods inside another. Why? Well, because you are throwing `self` to that method but you never recover the ownership again so you can not use `self` later. You could do something like:
```rust
self = self.secret(...); // now you are recovering the ownership
```
But this time, I prefer to simply modify the field.